src/analysis/special_functions/trigonometric/bounds.lean

```diff
@@ -0,0 +1,185 @@
+Authors: David Loeffler
+-/
+import analysis.special_functions.trigonometric.basic
+import analysis.special_functions.trigonometric.deriv
+import analysis.special_functions.trigonometric.arctan_deriv
+/-!
+# Polynomial bounds for trigonometric functions
+
+## Main statements
+
+This file contains upper and lower bounds for real trigonometric functions in terms
+of polynomials. See `trigonometric.basic` for more elementary inequalities, establishing
+the ranges of these functions, and their monotonicity in suitable intervals.
+
+Here we prove the following:
+
+* `sin_lt`: for `x > 0` we have `sin x < x`.
+* `sin_gt_sub_cube`: For `0 < x ≤ 1` we have `sin x > x  - x^3 / 4`.
+* `lt_tan`: for `0 < x < π/2` we have `x < tan x`.
+
+## Tags
+
+sin, cos, tan, angle
+-/
+
+noncomputable theory
+open set
+
+namespace real
+notation `π` := real.pi
+
+/-- For 0 < x, we have sin x < x. -/
+lemma sin_lt {x : ℝ} (h : 0 < x) : sin x < x :=
+begin
+  cases le_or_gt x 1 with h' h',
+  { have hx : |x| = x := abs_of_nonneg (le_of_lt h),
+    have : |x| ≤ 1, rwa [hx],
+    have := sin_bound this, rw [abs_le] at this,
+    have := this.2, rw [sub_le_iff_le_add', hx] at this,
+    apply lt_of_le_of_lt this, rw [sub_add], apply lt_of_lt_of_le _ (le_of_eq (sub_zero x)),
+    apply sub_lt_sub_left, rw [sub_pos, div_eq_mul_inv (x ^ 3)], apply mul_lt_mul',
+    { rw [pow_succ x 3], refine le_trans _ (le_of_eq (one_mul _)),
+      rw mul_le_mul_right, exact h', apply pow_pos h },
+    norm_num, norm_num, apply pow_pos h },
+  exact lt_of_le_of_lt (sin_le_one x) h'
+end
+
+/-- For 0 < x ≤ 1 we have sin x ≥ x  - x^3 / 4.
+
+This is also true for x > 1, but it's nontrivial for x just above 1. This inequality is not
+tight; the tighter inequality is sin x > x - x ^ 3 / 6, but this inequality has a simpler
+proof. -/
+lemma sin_gt_sub_cube {x : ℝ} (h : 0 < x) (h' : x ≤ 1) : x - x ^ 3 / 4 < sin x :=
+begin
+  have hx : |x| = x := abs_of_nonneg (le_of_lt h),
+  have : |x| ≤ 1, rwa [hx],
+  have := sin_bound this, rw [abs_le] at this,
+  have := this.1, rw [le_sub_iff_add_le, hx] at this,
+  refine lt_of_lt_of_le _ this,
+  rw [add_comm, sub_add, sub_neg_eq_add], apply sub_lt_sub_left,
+  apply add_lt_of_lt_sub_left,
+  rw (show x ^ 3 / 4 - x ^ 3 / 6 = x ^ 3 * 12⁻¹,
+    by simp [div_eq_mul_inv, ← mul_sub]; norm_num),
+  apply mul_lt_mul',
+  { rw [pow_succ x 3], refine le_trans _ (le_of_eq (one_mul _)),
+    rw mul_le_mul_right, exact h', apply pow_pos h },
+  norm_num, norm_num, apply pow_pos h
+end
+
+
+/- The next lemmas are building up to proving tan(x) > x for x ∈ (0,π/2). -/
+
+private def tan_minus_id (x : ℝ) : ℝ := tan x - x
+
+private def tansq (x : ℝ) (h : cos x ≠ 0) : ℝ := 1/(cos x)^2 - 1
+
+private lemma tan_minus_id_deriv (x : ℝ) (h: cos x ≠ 0) :
+    deriv tan_minus_id x = (tansq x h) :=
+begin
+  apply has_deriv_at.deriv,
+  simp only [tansq],
+  have uv := has_deriv_at.add (has_deriv_at_tan h) (has_deriv_at.neg (has_deriv_at_id x)),
+  simp at *,
+  exact uv,
```

Comment:
```suggestion
  apply has_deriv_at.deriv,
  simpa [tansq] using (has_deriv_at_tan h).add (has_deriv_at_id x).neg
```

note that this is not purely stylistic - you're not allowed to have `simp`s that don't finish the goal, because they could break unexpectedly if the simp lemmas change. the tactic `squeeze_simp` is helpful for this

also note the dot notation! it makes things a lot easier to type